### PR TITLE
allow dots in projects and tags

### PR DESCRIPTION
**Pull request by tolry**
reference to https://github.com/DavidBadura/doThings/issues/37

### CODE REVIEW

**Comment by tolry** [comment]
should I also add some tests for tags including dots?

File: tests/TaskManagerTest.php [hunk 1]
```diff
@@ -563,6 +564,21 @@ public function testTagNameNotAllowed()
...
+    public function testProjectHierarchyAllowed()
+    {
+        $projects = ['grandparent.parent.child', 'parent.child'];
+        foreach ($projects as $project) {
+            $task = new Task();
+            $task->setDescription('foo1');
+            $task->setProject($project);
+
+            // now exception thrown
+            $this->taskManager->save($task);
```
Comment (DavidBadura) [inline tests/TaskManagerTest.php]: you test only the write operation and not read. you should try fetch a new instance and check the project.

``` php
$this->taskManager->clear();
$task = $this->taskManager->find($task->getUuid);
// ...
```

Comment (DavidBadura) [inline tests/TaskManagerTest.php]: maybe it's easier to use a data provider: https://github.com/DavidBadura/Taskwarrior/blob/master/tests/TaskManagerTest.php#L86

Comment (tolry) [inline tests/TaskManagerTest.php]: Thanks, forgot about the identity map here. With or without DataProviders would probably not really save code here, but I can make the changes, sure.

**Comment by tolry** [comment]
@DavidBadura done

**Comment by DavidBadura** [comment]
thank you!

**Pull request merged**